agent/hcp/bootstrap/bootstrap.go

```diff
@@ -298,21 +298,23 @@ func persistAndProcessConfig(dataDir string, devMode bool, bsCfg *hcpclient.Boot
 			return "", fmt.Errorf("failed to persist bootstrap config: %w", err)
 		}
 
-		if err := validateManagementToken(bsCfg.ManagementToken); err != nil {
-			return "", fmt.Errorf("invalid management token: %w", err)
-		}
-		if err := persistManagementToken(dir, bsCfg.ManagementToken); err != nil {
-			return "", fmt.Errorf("failed to persist HCP management token: %w", err)
+		if bsCfg.ManagementToken != "" {
```

Comment:
As I understand, this works to handle our use-case!

I noticed in `validateManagementToken()` that there's an additional check for empty token string again [here](https://github.com/hashicorp/consul/blob/main/agent/hcp/bootstrap/bootstrap.go#L357-L359). It kind of contradicts the check here which might be confusing later on. What are your thoughts on removing the empty string check in validateManagementToken() and/or adding a comment to clarify that validation should be called only for tokens that exist?

Also maybe a brief comment here to explain why we would expect no management token and that it's okay?

Let me know what you think!